service/history/queuev2/virtual_slice.go

```diff
@@ -0,0 +1,348 @@
+// The MIT License (MIT)
```

Comment:
Please add a implementation detail oriented doc under docs/ folder with a component diagram showing their relationship etc. This would help with reviewing going forward